app/src/main/java/com/mapbox/services/android/navigation/testapp/example/ui/ExampleActivity.kt

```diff
@@ -0,0 +1,286 @@
+package com.mapbox.services.android.navigation.testapp.example.ui
+
+import android.arch.lifecycle.ViewModelProviders
+import android.content.Intent
+import android.location.Location
+import android.os.Bundle
+import android.support.design.widget.BottomSheetBehavior
+import android.support.transition.TransitionManager
+import android.support.v7.app.AppCompatActivity
+import android.view.View
+import android.widget.Toast
+import com.mapbox.android.core.permissions.PermissionsManager
+import com.mapbox.api.directions.v5.models.DirectionsRoute
+import com.mapbox.api.geocoding.v5.models.CarmenFeature
+import com.mapbox.geojson.Point
+import com.mapbox.mapboxsdk.camera.CameraUpdate
+import com.mapbox.mapboxsdk.camera.CameraUpdateFactory
+import com.mapbox.mapboxsdk.geometry.LatLngBounds
+import com.mapbox.mapboxsdk.maps.AttributionDialogManager
+import com.mapbox.mapboxsdk.maps.MapboxMap
+import com.mapbox.services.android.navigation.testapp.NavigationSettingsActivity
+import com.mapbox.services.android.navigation.testapp.R
+import com.mapbox.services.android.navigation.testapp.example.ui.autocomplete.AutoCompleteBottomSheetCallback
+import com.mapbox.services.android.navigation.testapp.example.ui.autocomplete.ExampleAutocompleteAdapter
+import com.mapbox.services.android.navigation.testapp.example.ui.permissions.PermissionRequestDialog
+import com.mapbox.services.android.navigation.testapp.example.utils.hideKeyboard
+import com.mapbox.services.android.navigation.testapp.example.utils.showKeyboard
+import com.mapbox.services.android.navigation.ui.v5.map.NavigationMapboxMap
+import com.mapbox.services.android.navigation.v5.navigation.MapboxNavigation
+import kotlinx.android.synthetic.main.activity_example.*
+
+class ExampleActivity : AppCompatActivity(), ExampleView {
+
+  private val permissionsManager = PermissionsManager(this)
+  private lateinit var presenter: ExamplePresenter
```

Comment:
What about?

```kotlin
private val presenter by lazy(mode = LazyThreadSafetyMode.NONE) {
  val viewModel = ViewModelProviders.of(this).get(ExampleViewModel::class.java)
  ExamplePresenter(this, viewModel)
}
```

This way `presenter` would be `val` (read-only) and also we'd remove the usage of `lateinit` - which we should treat carefully.